python/src/trezorlib/ui.py

```diff
@@ -172,6 +172,52 @@ def get_passphrase(self, available_on_device: bool) -> Union[str, object]:
                 raise Cancelled from None
 
 
+class ScriptUI:
+    """Interface to be used by scripts, not directly by user.
+
+    Communicates with a client application using print() and input().
+
+    Lot of `ClickUI` logic is outsourced to the client application, which
+    is responsible for supplying the PIN and passphrase.
+
+    Reference client implementation can be found under `cli/test_script_client.py`.
+    """
+
+    def __init__(self) -> None:
```

Comment:
As `self` is not being used in any method, can it be somehow simplified?

Reply:
you can make everything a `@staticmethod` and pass in `ScriptUI` class object instead of `ScriptUI()` instance

Reply:
Good idea. Unfortunately, from type-checking view it is problematic, as `def get_ui(self) -> TrezorClientUI:` signature is invalid if we would return just `ScriptUI` (in addition to `ClickUI(...)`) - once we return class, once instance.

(_Maybe_ it is not an issue, as running  the newest `pyright` (`1.1.204`) does not see any problems there. In `Pylance` - probably having an older `pyright` version - it is flagged as an issue)

Reply:
yeah, this is both correct and [already fixed](https://github.com/microsoft/pyright/issues/2729)

Reply:
Changed in e5670a4